cloud/terraform/terraform.py

```diff
@@ -45,39 +44,45 @@ def __new_aws_provider(self, region):
             'profile': self.aws_profile,
         }
 
-        return new_provider
+        self.providers['provider'][self.cloud].append(new_provider)
+
+    def add_instance_conf(self, instance):
+        if self.cloud == 'aws':
+            self.__new_aws_instance(instance)
+
+    def __new_aws_instance(self, instance):
+        if not instance['instance_type']:
+            instance['instance_type'] = 't2.micro'
+
+        name = instance['name'].replace('.', '-')
 
-    def add_instance_conf(self, region, image_id, name, instance_type=''):
+        # TODO: agree on a name for 'region' or 'location' across all cloud so we can move
+        #      this code to 'add_instance_conf'
         aliases = [provider['alias'] for provider in self.providers['provider'][self.cloud]]
-        if region not in aliases:
+        if instance['region'] not in aliases:
             print('Cannot add an instance if region provider is not set up')
             exit(1)
 
-        if self.cloud == 'aws':
-            self.__new_aws_instance(region, image_id, name, instance_type)
-
-    def __new_aws_instance(self, region, image_id, name, instance_type):
-        if not instance_type:
-            instance_type = 't2.micro'
-
-        resource_id = f'{region}-{name}'.replace('.', '-')
         new_instance = {
-            'instance_type': instance_type,
-            'ami': image_id,
-            'provider': f'aws.{region}',
+            'instance_type': instance['instance_type'],
+            'ami': instance['ami'],
+            'provider': f'aws.{instance["region"]}',
             'tags': {'name': name},
         }
         self.resources['resource']['aws_instance'][name] = new_instance
 
     def print_configuration(self):
-        print(
-            '------------\nResources:',
-            str(self.resources),
-            '\n------------\nProviders:',
-            str(self.providers),
-        )
+        pprint(self.resources)
+        pprint(self.providers)
 
     def set_configuration(self):
+        main_tf = {
+            'terraform': {
+                'required_providers': {'aws': {'source': 'hashicorp/aws', 'version': '~> 3.27'}},
```

Comment:
I guess we can decouple this part from here and add it programmatically, when we introduce a new cloud.

Or we can add the logic now to set the provider depending on `self.cloud`.

What do you think?